### PR TITLE
chore(backend): Refactor `copy_from` method to be more generic

### PR DESCRIPTION
**Pull request by amanape**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
From `zip_files_in_sandbox` to `copy_from`

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Review by xingyaoww — COMMENTED** [review]
I wonder if it is possible to implement this on the runtime-client side, similar to `copy_to`, so we can also support this easily on RemoteRuntime as well?


https://github.com/All-Hands-AI/OpenHands/blob/59fbc11afeb4a7d9dd3ae4dd2a7e05b34421dd8b/openhands/runtime/client/client.py#L709-L761

**Comment by amanape** [comment]
Good idea, I'll do that

File: openhands/runtime/client/runtime.py [hunk 1]
```diff
@@ -557,9 +557,11 @@ def list_files(self, path: str | None = None) -> list[str]:
         except Exception as e:
             raise RuntimeError(f'List files operation failed: {str(e)}')
 
-    def zip_files_in_sandbox(self) -> bytes:
-        """Zips the files in the sandbox and returns the bytes for streaming."""
-        sandbox_dir = os.getcwd() + self.config.workspace_mount_path_in_sandbox
+    def copy_from(self, src: str | None = None) -> bytes:
```
Comment (rbren) [inline openhands/runtime/client/runtime.py:560]: @xingyaoww there's another runtime where we need to implement this, right?

Comment (xingyaoww) [reply openhands/runtime/client/runtime.py:560]: yes - i think we at least need to do this for `EventStream` and `RemoteRuntime`

File: openhands/runtime/client/client.py [hunk 2]
```diff
@@ -711,7 +714,8 @@ async def upload_file(
         file: UploadFile, destination: str = '/', recursive: bool = False
     ):
         assert client is not None
-
+        print('CLIENT AVAILABLE' if client is not None else 'CLIENT UNAVAILABLE')
+        print('DESTINATION: ', destination)
```
Comment (xingyaoww) [inline openhands/runtime/client/client.py:718]: Can we replace this with logger.debug? Or remove it directly

**Review by xingyaoww — COMMENTED** [review]
Mostly LGTM! Can we also add a few test cases for this function?

File: openhands/runtime/client/client.py (hunk 2 above)
Comment (amanape) [reply openhands/runtime/client/client.py]: Woops! Leftovers

**Review by rbren — APPROVED** [review]

**Pull request merged**